packages/core/src/validate/index.ts

```diff
@@ -41,7 +41,12 @@ export function validate(
       parse(`
           ${doc.source.body}
 
-          ${fragments.map(print).join('\n\n')}
+          ${fragments
+            .filter(
+              f => doc.source.body.indexOf(`fragment ${f.name.value}`) < -1,
```

Comment:
I noticed `< -1`, maybe you meant `> -1`?

Fragment named `foo-bar` will pass the check even if `foo-bar-baz` was defined. Maybe we should add `fragment ${f.name.value} on`

Reply:
`< -1` was intentional because I was filtering out the ones which are in the document, but it also works without < -1, so put that now. 

Added `fragment ${f.name.value} on`.